llumnix/llumlet/local_migration_scheduler.py

```diff
@@ -18,20 +18,36 @@
 from llumnix.backends.backend_interface import BackendInterface
 
 class LocalMigrationScheduler:
-    def __init__(self, request_migration_policy: str, backend_engine: BackendInterface) -> None:
+    def __init__(self, request_migration_policy: str, backend_engine: BackendInterface, strict_pre_migration: bool) -> None:
         self.request_migration_policy = request_migration_policy
         self.backend_engine = backend_engine
+        self.strict_pre_migration = strict_pre_migration
 
     def get_migrate_out_request(self, min_request_len=0, max_request_len=np.inf) -> Optional[LlumnixRequest]:
-        migrate_out_request: LlumnixRequest = None
-        if self.request_migration_policy == 'LCFS':
-            migrate_out_request = self.get_last_running_request(min_request_len, max_request_len)
-        elif self.request_migration_policy == 'LJF':
-            migrate_out_request = self.get_longest_running_request(min_request_len, max_request_len)
-        elif self.request_migration_policy == 'SJF':
-            migrate_out_request = self.get_shortest_running_request(min_request_len, max_request_len)
+         # Requests meet the strict pre-migration always have higher prioirity than other migration policy.
+        migrate_out_request = self.get_ready_migration_request(min_request_len, max_request_len)
```

Comment:
make `get_ready_migration_request` a migration_policy?

Reply:
> make `get_ready_migration_request` a migration_policy?

The `get_ready_migration_request` cannot be treated as a user-optional policy like other migration policies. The priority of migration requests that have reached `expected_steps` should be the highest.